gcs/client.go

```diff
@@ -0,0 +1,61 @@
+package gcs
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	gcStorage "cloud.google.com/go/storage"
+)
+
+type Client interface {
+	Read(ctx context.Context) ([]byte, error)
+	Write(ctx context.Context, p []byte) error
+}
+
+type Adapter struct {
+	config Config
+	client *gcStorage.Client
+}
+
+func (a Adapter) Read(ctx context.Context) ([]byte, error) {
+	r, err := a.client.Bucket(a.config.Bucket).Object(a.config.Name).NewReader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+
+	body, err := ioutil.ReadAll(r)
```

Comment:
The `ioutil.ReadAll()` method has been deprecated in Go 1.16. Use `io.ReadAll()` instead.
https://pkg.go.dev/io/ioutil#ReadAll